app/src/main/java/tech/zafrani/companionforpubg/utils/PUBGNewsFetch.java

```diff
@@ -0,0 +1,65 @@
+package tech.zafrani.companionforpubg.utils;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import java.io.IOException;
+
+import tech.zafrani.companionforpubg.adapters.NewsAdapter;
+import tech.zafrani.companionforpubg.models.NewsItem;
+
+public class PUBGNewsFetch extends AsyncTask<Void, NewsItem, Void>  {
+    private final NewsAdapter adapter;
+
+    public PUBGNewsFetch(@NonNull final NewsAdapter adapter) {
```

Comment:
This is good. However now PUBGNewsFetch forces the class that uses it to provide an adapter. There may be cases where an adapter isn't needed. To simplify things you can instead define a Listener interface inside of PUBGNewsFetch which contains an update method that receives News. Make the fragment implement that interface and update the adapter there.